scipy/stats/_hypotests.py

```diff
@@ -130,3 +131,133 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
     p = chi2.sf(w, r)
 
     return Epps_Singleton_2sampResult(w, p)
+
+
+def poisson_etest(k1, k2, n1, n2, diff=0, alternative='two-sided'):
+    """
+    Calculate E-test for the mean difference of two samples
+
+    The test requires two samples coming from poisson distribution. This is a
+    right-sided and two-sided test.
+
+    Parameters
+    ----------
+    k1, k2 : int or float
+        Count from the first and second samples respectively
+    n1, n2 : int or float
+        Sample size of each sample
+    diff : int of float, optional
```

Comment:
remove float

Reply:
But for `n1`, `n2`, and `diff`, I am not so sure. Take for example in docstring, where `n` is measured in gram (possible to have float values). Also study from Stampfer and Willett (1985). A prospective study to examine the relationship of post-menopausal hormone use and coronary heart disease (CHD). With postmenopausal hormone use in 54308.7 person-years, there are 30 CHD cases; without postmenopausal hormone use in 51477.5 person-years, there are 60 CHD cases. This would imply that `n1` and `n2` can be float right?

Or do we need to change the argument name and description to something different. Link given by @josef-pkt above refer to another poisson test implementation using `exposure` instead of `n` that has description "Total exposure (time * subjects) in the sample"

Reply:
yeah, that's possible. It is the example that makes me unsure, using gram unit which supposed to be a float. But since the formula also used summation (signifying `n_i` is an int), let stick with it

Reply:
do you intend to change that to int?

Reply:
I think there is confusion here that is due to the choice of "unit". The lambda parameter is a rate of occurrences per some unit. We are free to choose that unit to be whatever we want, and the corresponding rate will change accordingly. 

I think 
> But since the formula also used summation
refers to
![image](https://user-images.githubusercontent.com/6570539/129458762-d9b3fecb-a379-41c9-877e-79dd9c27421d.png)

An observation is the number of occurrences observed in a given unit, whatever size that unit may be. We could have chosen smaller units, however, and then that sum starts to look like integration. 

I don't have a rigorous proof, but based on that idea and looking at the form of the equations below, I think it is safe to let `n1` and `n2` be floats. I think this line of reasoning can also explain why we get the same p-value regardless of whether we multiply `n1` and `n2` by the same factor (answering @chrisb83's [question](https://github.com/scipy/scipy/pull/10561#discussion_r363919362).

Reply:
Matt, that's a fair point to make. But `n1` and `n2` in my opinion should represent how many observation is taken, like your examples [below](https://github.com/scipy/scipy/pull/10561#discussion_r688993386) not a unit metric

Regarding the parameter names, i don't have any comment (sorry lol). Perhaps @chrisb83 have any take on this? since the initial change is to align with `ttest_ind_from_stats`

Reply:
@mdmahendri I'm not sure I understand your position on this - have you changed your mind since your comments [above](https://github.com/scipy/scipy/pull/10561#discussion_r310441968) suggesting that these arguments could accept floats?